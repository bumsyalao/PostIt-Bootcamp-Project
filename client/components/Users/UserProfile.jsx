import React, { Component } from 'react';
import { connect } from 'react-redux';
import { allUserGroups } from '../../actions/users';
import photo from '../../images/photo.jpg';

/**
 * @class UserProfile
 * @extends {Component}
 */
export class UserProfile extends Component {

  /**
   * Creates an instance of UserProfile.
   * Binds class methods
   * @memberOf UserProfile
   */
  constructor() {
    super();
    this.state = {
      user: []
    };
  }

  /**
   * Makes action call to get UserId
   * @returns userId
   * @memberOf UserProfile
   */
  componentDidMount() {
    const userid = this.props.access.user.userId;
    this.props.allUserGroups(userid);
  }

  /**
   * Update the state if the props are changed
   * @param {object} nextProps
   * @memberOf UserProfile
   */
  componentWillReceiveProps(nextProps) {
    this.setState({
      groups: nextProps.user,
      user: nextProps.access.user
    });
  }
  /**
   * Renders UserProfile
   * @returns UserProfile
   * @memberOf UserProfile
   */
  render() {
    return (
      <div className="row">
        <div className="container form-margin">
          <div className="col s12 m7">
            <div className="profile-image">
              <img
                src={photo}
                width="100px"
                height="100px"
              />
            </div>
            <div className="card-content">
              <div className="collection">
                <div className="collection-item">
                  <span>{this.props.access.user.username} </span>
                </div>
                <div className="collection-item">
                  <span>{this.props.access.user.email} </span>
                </div>
                <div className="collection-item">
                  <span>{this.props.access.user.phoneNumber} </span>
                </div>
                <div className="collection-item">
                  {' '}
                  Groups I belong to:{' '}
                  {this.state.groups &&
                    this.state.groups.map(group => (
                      <div className="chip" key={group.groupId} > {group.groupName}
                      </div>
                    ))}{' '}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  access: state.access,
  user: state.users.usergroups
});

export default connect(mapStateToProps, { allUserGroups })(UserProfile);
