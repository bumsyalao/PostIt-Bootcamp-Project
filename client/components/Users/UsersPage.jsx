import React, { Component } from 'react';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';
import { getAllUsers } from '../../actions/users';

class UsersPage extends Component {
  constructor() {
    super();
    this.state = {
      users: [],
      offset: 0,
      pageCount: 0,
      limit: 5,
      searchParam: ''
    };
    this.handlePageClick = this.handlePageClick.bind(this);
  }

  componentWillMount() {
    this.getUsers();
  }

  getUsers(offset, limit) {
    this.props.getAllUsers(offset, limit, this.state.searchParam);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      users: nextProps.users.users,
      pageCount: nextProps.pagination.pageCount,
      count: nextProps.pagination.count });
  }

  searchUsers = (e) => {
    this.setState({ searchParam: e.target.value, offset: 0, limit: 5 });
  }

  onSearch = () => {
    const limit = 5, offset = 0;
    if (!this.state.searchParam) return;
    this.getUsers(offset, limit);
  }
  handlePageClick(data) {
    const selected = data.selected;
    const limit = 5;
    const offset = Math.ceil(selected * limit);
    this.setState({ offset });
    this.getUsers(offset, limit).then(() => {
      this.setState({
        users: this.props.users.users
      });
    });
  }

  render() {
    const { users } = this.state;
    return (
      <div className="row">
        <div className="container form-margin">
          <div className="">
            <div className="row">
              <form className="col s12" >
                <div className="row">
                  <div className="input-field col s12">
                    <i className="material-icons prefix">search</i>
                    <input id="icon_prefix"
                          type="text"
                          onChange={this.searchUsers}
                          className="validate" />
                    <label htmlFor="icon_prefix">Enter username</label>
                    <a className="waves-effect waves-light red lighten-2 btn"
                        onClick={this.onSearch}>
                        Search
                    </a>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>

        <div className="container table">
          <table className="responsive-table">
            <thead>
              <tr>
                <th>USERNAME</th>
                <th>EMAIL</th>
              </tr>
            </thead>
            <tbody>
                {users && users.map(({ email, username }) =>
                  <tr>
                    <td>{username}</td>
                    <td>{email}</td>
                  </tr>
              )}
            </tbody>
          </table>
        {(this.state.count > 5) && <ReactPaginate
                      previousLabel={'previous'}
                      nextLabel={'next'}
                      pageCount={this.state.pageCount}
                      marginPagesDisplayed={1}
                      pageRangeDisplayed={3}
                      onPageChange={this.handlePageClick}
                      containerClassName={'pagination'}
                      subContainerClassName={'pages pagination'}
                      activeClassName={'active'}
                    />}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  users: state.users,
  pagination: state.users.pagination
});


export default connect(mapStateToProps, { getAllUsers })(UsersPage);
