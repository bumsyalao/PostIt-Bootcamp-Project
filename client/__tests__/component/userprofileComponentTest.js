/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
import { UserProfile } from '../../components/Users/UserProfile';

configure({ adapter: new adapter() });

describe('UserProfile Compoenent', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    access: {
      user: { id: 1 }
    },

    allUserGroups: jest.fn(() => Promise.resolve())
  };
  const nextProps = {
    user: spy,
    access: {
      user: spy
    }
  };
  const component = shallow(
    <UserProfile {...props} />
  );

  test('should render componemt', () => {
    expect(component.exists()).toBe(true);
  });

  test('should match User Profile component snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
