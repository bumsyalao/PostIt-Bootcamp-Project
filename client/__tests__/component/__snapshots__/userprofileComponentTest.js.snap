// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Compoenent should match User Profile component snapshot 1`] = `
<div
  className="row"
>
  <div
    className="container form-margin"
  >
    <div
      className="col s12 m7"
    >
      <div
        className="profile-image"
      >
        <img
          height="100px"
          src={
            Object {
              "process": [Function],
            }
          }
          width="100px"
        />
      </div>
      <div
        className="card-content"
      >
        <div
          className="collection"
        >
          <div
            className="collection-item"
          >
            <span>
               
            </span>
          </div>
          <div
            className="collection-item"
          >
            <span>
               
            </span>
          </div>
          <div
            className="collection-item"
          >
            <span>
               
            </span>
          </div>
          <div
            className="collection-item"
          >
             
            Groups I belong to:
             
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
